File: src/main/java/com/mpe85/grampa/visitor/RuleVisitor.java
package com.mpe85.grampa.visitor;

import com.mpe85.grampa.rule.Rule;
import com.mpe85.grampa.rule.impl.AbstractRule;

/**
 * Visitor for {@link Rule}s.
 * 
 * @author mpe85
 *
 * @param <T>
 *        type of the parser stack values
 */
public interface RuleVisitor<T> {
	
	/**
	 * Visits an abstract rule.
	 * 
	 * @param rule
	 *        the rule to visit
	 */
	void visit(AbstractRule<T> rule);
	
}

File: src/main/kotlin/com/mpe85/grampa/visitor/RuleVisitor.kt
package com.mpe85.grampa.visitor

import com.mpe85.grampa.rule.Rule
import com.mpe85.grampa.rule.impl.AbstractRule

/**
 * Visitor for [Rule]s.
 *
 * @author mpe85
 *
 * @param[T] The type of the parser stack values
 */
public interface RuleVisitor<T> {

    /**
     * Visit an abstract rule.
     *
     * @param[rule] The rule to visit
     */
    public fun visit(rule: AbstractRule<T>)
}
